components/brave_wallet/browser/filecoin_keyring.cc

```diff
@@ -146,4 +142,15 @@ std::string FilecoinKeyring::GetAddressInternal(HDKeyBase* hd_key_base) const {
       .EncodeAsString();
 }
 
+absl::optional<std::string> FilecoinKeyring::SignTransaction(
+    const FilTransaction* tx) {
+  if (!tx)
+    return absl::nullopt;
+  auto address = tx->from().EncodeAsString();
+  HDKey* hd_key = static_cast<HDKey*>(GetHDKeyFromAddress(address));
+  if (!hd_key)
+    return absl::nullopt;
+  return tx->GetSignedTransaction(base::Base64Encode(hd_key->private_key()));
```

Comment:
I'm still learning the C++ memory model here, so this is a bit less clear if it's wiping the memory off the stack now once this function scope closes. @diracdeltas or @fmarier can you help me out here and double check to make sure we end up getting the right thing for this?

From what it looks like to me we're passing a pointer into the `Base64Encode` function to get the value returned from `private_key()` and then passing that into `GetSignedTransaction`. It's not clear to me if this means that the return value of this `Base64Encode` function is executing only via stack memory here or if it may end up on the heap now because it's getting a pointer from the `hd_key` memory which I assume is coming from the heap.

Reply:
probably a better question for @bridiver

Reply:
@spylogsster this is the last one I want to get double checked, but once it's done I think this PR is good to go

Reply:
@bridiver sent me a message on slack to explain the memory model here and I don't believe this is a problem anymore. I think the browser code here is alright